src/macros/@netput.jl

```diff
@@ -0,0 +1,228 @@
+__NETPUT_KWARGS__ = [:taxes, :reference_price]
+
+struct PreNetput
+    commodity::Any
+    quantity::Any
+    nest::Any
+    taxes::Any
+    reference_price::Any
+    netput_type::Any
+    index_vars::Vector{Any} # Index variables for the commodity
+end
+
+
+
+# Used and safe
+function parse_netput_macro_arguments(
+    error_fn::Function, 
+    input_args,
+    netput_fn::DataType; 
+    num_positional_args = 3, 
```

Comment:
commodity, quantity and nest name are the 3 required arguments (no more, no less)